server.go

```diff
@@ -10,52 +10,64 @@ import (
 type subscription struct {
 	channel     string
 	lastEventID string
-	out         chan interface{}
+	out         chan<- eventOrComment
 }
 
 type eventOrComment interface{}
 
 type outbound struct {
 	channels       []string
 	eventOrComment eventOrComment
+	ackCh          chan<- struct{}
 }
 
 type registration struct {
 	channel    string
 	repository Repository
 }
 
+type unregistration struct {
+	channel         string
+	forceDisconnect bool
+}
+
 type comment struct {
 	value string
 }
 
+type eventBatch struct {
+	events <-chan Event
+}
+
 // Server manages any number of event-publishing channels and allows subscribers to consume them.
 // To use it within an HTTP server, create a handler for each channel with Handler().
 type Server struct {
-	AllowCORS     bool          // Enable all handlers to be accessible from any origin
-	ReplayAll     bool          // Replay repository even if there's no Last-Event-Id specified
-	BufferSize    int           // How many messages do we let the client get behind before disconnecting
-	Gzip          bool          // Enable compression if client can accept it
-	MaxConnTime   time.Duration // If non-zero, HTTP connections will be automatically closed after this time
-	Logger        Logger        // Logger is a logger that, when set, will be used for logging debug messages
-	registrations chan *registration
-	pub           chan *outbound
-	subs          chan *subscription
-	unregister    chan *subscription
-	quit          chan bool
-	isClosed      bool
-	isClosedMutex sync.RWMutex
+	AllowCORS       bool          // Enable all handlers to be accessible from any origin
+	ReplayAll       bool          // Replay repository even if there's no Last-Event-Id specified
+	BufferSize      int           // How many messages do we let the client get behind before disconnecting
+	Gzip            bool          // Enable compression if client can accept it
+	MaxConnTime     time.Duration // If non-zero, HTTP connections will be automatically closed after this time
+	Logger          Logger        // Logger is a logger that, when set, will be used for logging debug messages
+	registrations   chan *registration
+	unregistrations chan *unregistration
```

Comment:
The only new field here is `unregistrations`. I also renamed the `unregister` channel to `unsubs`, because that's what it is (it is the opposite of `subs`, not of `registrations`).